summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormPlatform.scala

```diff
@@ -184,19 +187,17 @@ abstract class Storm(options: Map[String, Options], transformConfig: Summingbird
 
     val metrics = getOrElse(stormDag, node, DEFAULT_SPOUT_STORM_METRICS)
 
-    val registerAllMetrics = new Function1[TopologyContext, Unit] {
-      def apply(context: TopologyContext) = {
-        // Register metrics passed in SpoutStormMetrics option.
-        metrics.metrics().foreach {
-          x: StormMetric[IMetric] =>
-            context.registerMetric(x.name, x.metric, x.interval.inSeconds)
-        }
-        // Register summingbird counter metrics.
-        StormStatProvider.registerMetrics(jobID, context, countersForSpout)
-        SummingbirdRuntimeStats.addPlatformStatProvider(StormStatProvider)
+    val registerAllMetrics = Externalizer({ context: TopologyContext =>
+      // Register metrics passed in SpoutStormMetrics option.
+      metrics.metrics().foreach {
+        x: StormMetric[IMetric] =>
+          context.registerMetric(x.name, x.metric, x.interval.inSeconds)
       }
-    }
-    val stormSpout = tormentaSpout.openHook(registerAllMetrics).getSpout
+      // Register summingbird counter metrics.
+      StormStatProvider.registerMetrics(jobID, context, countersForSpout)
+      SummingbirdRuntimeStats.addPlatformStatProvider(StormStatProvider)
+    })
+    val stormSpout = tormentaSpout.openHook(registerAllMetrics.get).getSpout
```

Comment:
Was looking around this today; it's possible this doesn't cause an issue right now because the default `Spout` implementation for openHook is to ignore it:

https://github.com/twitter/tormenta/blob/v0.11.1/tormenta-core/src/main/scala/com/twitter/tormenta/spout/Spout.scala#L49

I ran into issues with this during some testing I was doing, probably because I was using `BaseSpout` which actually keeps the hook:

https://github.com/twitter/tormenta/blob/v0.11.1/tormenta-core/src/main/scala/com/twitter/tormenta/spout/BaseSpout.scala#L32

Maybe `BaseSpout` could do the externalizing, but it seems like all spout implementations (that do open hooks) would need to as well.